container/container.go

```diff
@@ -133,14 +137,18 @@ func ConfigForTask(taskInfo *mesos.TaskInfo) *docker.CreateContainerOptions {
 
 	// Check for and calculate CPU shares
 	cpus := getResource("cpus", taskInfo)
-	if cpus != nil {
-		config.Config.CPUShares = int64(*cpus.Scalar.Value * float64(1024))
+	if cpus != nil && forceCpuLimit {
+		config.HostConfig.CPUPeriod = defaultCpuPeriod
+		config.HostConfig.CPUQuota = int64(*cpus.Scalar.Value * float64(defaultCpuPeriod))
+		log.Infof("CPU limit set [HostConfig.CPUQuota=%d, HostConfig.CPUPeriod=%d]  ", config.HostConfig.CPUQuota, defaultCpuPeriod)
 	}
 
 	// Check for and calculate memory limit
-	memory := getResource("memoryMb", taskInfo)
-	if memory != nil {
-		config.Config.Memory = int64(*memory.Scalar.Value * float64(1024*1024))
+	memory := getResource("mem", taskInfo)
```

Comment:
changed `memoryMb` for `mem`. Somehow we were not using the right resource name.